### PR TITLE
ott-mode: Factor our from ott

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

> ott-mode: Init (version inherited from ott)
>
> This way it is pre-compiled.

>  ott: Don't also install emacs mode
>
> There is now a separate package for that.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/logic/ott/default.nix [hunk 1]
```diff
@@ -16,7 +16,13 @@ stdenv.mkDerivation rec {
...
   installTargets = "ott.install";
 
-  postInstall = "opaline -prefix $out";
+  postInstall = ''
+    opaline -prefix $out
+  ''
+  # There is `emacsPackages.ott-mode` for this now.
+  + ''
+    rm -r $out/share/emacs
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/ott/default.nix:25]:
```suggestion
    opaline -prefix $out

    # There is `emacsPackages.ott-mode` for this now.
    rm -r $out/share/emacs
  '';
```

Comment (Ericson2314) [reply pkgs/applications/science/logic/ott/default.nix:25]: I prefer to always avoid comments going in the derivations. It causes more rebuilds and wastes space.

Comment (SuperSandro2000) [reply pkgs/applications/science/logic/ott/default.nix:25]: This is causing a rebuild anyway and the comment does not get any bigger by this. Actually my suggestion saves a line.

Comment (Ericson2314) [reply pkgs/applications/science/logic/ott/default.nix:25]: Sorry I was a bit unclear. I mean it causes rebuilds if the comment alone needs to be changed in the future, and it wastes space in DRV files.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108507` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>ott</li>
</ul>
</details>

**Pull request merged**